scripts/BadgeManager.js

```diff
@@ -0,0 +1,54 @@
+// Handles the badge on the icon
+
+'use strict';
+
+function BadgeManager(addEventListener, isEnabled) {
+
+	addEventListener("hourMusic", function(hour) {
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: formatHour(hour) });
+		}
+	});
+	
+	addEventListener("weatherMusic", function(hour, music, weather) {
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: weather });
+		}
+	});
+
+	addEventListener("kkStart", function() {
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: "KK" });
+		}
+	});
+
+	addEventListener("pause", function() {
+		if(isEnabled()) {
+			chrome.browserAction.setBadgeText({ text: "" });
+		}
+	});
+	
+
+	chrome.browserAction.setBadgeBackgroundColor({ color: [57, 230, 0, 255] });
+}
+
+function clearBadge() {
```

Comment:
Are these functions meant to be in the global namespace? This breaks encapsulation, and I don't see where they're being used.